Header header
uint32 flag_state
